enterprise_gateway/services/sessions/kernelsessionmanager.py

```diff
@@ -174,39 +171,40 @@ def _delete_sessions(self, kernel_ids):
         try:
             for kernel_id in kernel_ids:
                 # Prior to removing session, update the per User list
-                kernel_session = self._sessions[kernel_id]
-                username = kernel_session['username']
-                if username in self._sessionsByUser and kernel_id in self._sessionsByUser[username]:
-                    self._sessionsByUser[username].remove(kernel_id)
-                self._sessions.pop(kernel_id, None)
-
-            self.save_sessions()  # persist changes
+                kernel_session = self._sessions.get(kernel_id, None)
+                if kernel_session is not None:
+                    username = kernel_session['username']
+                    if username in self._sessionsByUser and kernel_id in self._sessionsByUser[username]:
+                        self._sessionsByUser[username].remove(kernel_id)
+                    self._sessions.pop(kernel_id, None)
+
+            self.delete_sessions()
```

Comment:
Rather than store the kernel_ids in `self.sessions_to_remove`, I think we should just define the method to take a list of kernel_ids - just like `_delete_sessions` and avoid maintaining the `self.sessions_to_remove` member since `delete_sessions()` is only called from `_delete_sessions(kernel_ids)`.  I'm concerned that a member variable will be difficult to manage between calls since its essentially shared state.

Reply:
Done.